ledger-api/grpc-definitions/com/daml/ledger/api/v1/transaction_service.proto

```diff
@@ -128,6 +135,63 @@ message GetFlatTransactionResponse {
   Transaction transaction = 1;
 }
 
+message GetEventsByContractIdRequest {
+  // The contract id being queried
+  // Required
+  string contract_id = 1;
+
+  // The parties whose events the client expects to see.
+  // Events that are not visible for the parties in this collection will not be present in the response.
```

Comment:
Please use more precise language than "visible". Eg

Events on which at least one party form this set is an informee will be included. 
Events that are are only witnessed by one of the parties in the set are not included.

Reply:
Maybe even link to https://docs.daml.com/concepts/ledger-model/ledger-privacy.html#projections for a definition of Informee.

Reply:
IMO the query by ContractId, ContractKey should use the same visibility determination as GetEventById and GetTransactions and other methods from that family.

Reply:
The events are filtered by the witnesses of the event.  From the linked page "every action act on c is shown to all informees of all ancestor actions of act. These informees are called the witnesses of act".